mobile/library/swift/HeadersBuilder.swift

```diff
@@ -3,10 +3,9 @@ import Foundation
 private let kRestrictedPrefixes = [":", "x-envoy-mobile"]
 
 private func isRestrictedHeader(name: String) -> Bool {
-  let isHostHeader = name.caseInsensitiveCompare("host") == .orderedSame
-  lazy var hasRestrictedPrefix = kRestrictedPrefixes
-    .contains { name.range(of: $0, options: [.caseInsensitive, .anchored]) != nil }
```

Comment:
This fails to build with C++ interop enabled.

```
library/swift/HeadersBuilder.swift:8:22: error: value of type 'String' has no member 'range'
    .contains { name.range(of: $0, options: [.caseInsensitive, .anchored]) != nil }
                ~~~~ ^~~~~
library/swift/HeadersBuilder.swift:8:47: error: reference to member 'caseInsensitive' cannot be resolved without a contextual type
    .contains { name.range(of: $0, options: [.caseInsensitive, .anchored]) != nil }
                                              ^
library/swift/HeadersBuilder.swift:8:65: error: reference to member 'anchored' cannot be resolved without a contextual type
    .contains { name.range(of: $0, options: [.caseInsensitive, .anchored]) != nil }
                                                                ^
```

I wasn't able to find a way to keep the same code, so I refactored it to avoid Foundation APIs.